### PR TITLE
config(AWS): initialize configuration for AWS roles (TDE-176)

### PR DESCRIPTION
**Pull request by paulfouquet**
### Change Description:
Link AWS roles local configuration file to be used by `topo-processor`

### Notes for Testing:
N/A

#### Source Code Documentation Tasks:
- [x] README updated (where applicable)
- [ ] CHANGELOG (Unreleased section) updated
- [ ] Docstrings / comments included to help explain code

#### User Documentation Tasks:
- [ ] Confluence updated (where applicable)

#### Testing Tasks:
- [x] Added tests that fail without this change
- [x] All tests are passing in development environment
- [x] Reviewers assigned
- [ ] Linked to main issue for ZenHub board

### CODE REVIEW

File: topo_processor/util/aws_credentials.py [hunk 1]
```diff
@@ -1,29 +1,51 @@
...
+        default_credentials = Credentials(
+            session_credentials.access_key, session_credentials.secret_key, session_credentials.token
+        )
+
     return default_credentials
+
+
+def load_roles(roles_config: Dict) -> None:
+    for (key, value) in roles_config.items():
+        bucket_roles[bucket_name_from_path(key)] = {"role": value["roleArn"]}
```
Comment (blacha) [inline topo_processor/util/aws_credentials.py:51]: might be best to make this the same. eg use 

```
bucket_roles[bucket_name_from_path(key)] = value
```

so it's easy to reason about.

**Review by amfage — APPROVED** [review]

File: topo_processor/util/aws_credentials.py [hunk 2]
```diff
@@ -1,29 +1,55 @@
...
+session = boto3.Session(profile_name=os.getenv("AWS_PROFILE"))
+client: boto3.client = session.client("sts")
+bucket_roles: Dict[str, str] = {}
+assume_role_cache: Dict = {}
+
 
+def get_credentials(bucket_name: str) -> Credentials:
+    if not bucket_roles:
+        load_roles(json.load(open(aws_role_config_path)))
+    print(session.profile_name)
```
Comment (blacha) [inline topo_processor/util/aws_credentials.py:32]: print?

File: topo_processor/util/aws_credentials.py [hunk 3]
```diff
@@ -1,29 +1,55 @@
...
 
-def get_credentials(bucket_name: str):
+default_credentials: Credentials
+session = boto3.Session(profile_name=os.getenv("AWS_PROFILE"))
+client: boto3.client = session.client("sts")
+bucket_roles: Dict[str, str] = {}
+assume_role_cache: Dict = {}
+
 
+def get_credentials(bucket_name: str) -> Credentials:
```
Comment (blacha) [inline topo_processor/util/aws_credentials.py:29]: these credentials need to be refreshed somewhere, was that taken out of this PR?

Comment (paulfouquet) [reply topo_processor/util/aws_credentials.py:29]: I've already spent a lot of more time that I've expected on this. I would prefer do that in another PR. I've created https://linzgovt.atlassian.net/browse/TDE-235

**Review by blacha — APPROVED** [review]

**Review by blacha — APPROVED** [review]

**Pull request merged**